src/v/compression/stream_zstd.cc

```diff
@@ -46,18 +67,16 @@ stream_zstd::zstd_compress_ctx& stream_zstd::compressor() {
     return _compress;
 }
 
-void stream_zstd::reset_decompressor() {
-    _decompress.reset(ZSTD_createDCtx());
-    if (!_decompress) {
-        throw std::bad_alloc{};
-    }
-}
-
-stream_zstd::zstd_decompress_ctx& stream_zstd::decompressor() {
-    if (!_decompress) {
-        reset_decompressor();
+ZSTD_DCtx* stream_zstd::decompressor() {
+    if (unlikely(!dctx_workspace)) {
+        /*
+         * handle on-demand case for tests, etc..
+         */
+        init_workspace(2_MiB);
```

Comment:
don't you need to pass the `property<size_t> workspace_size` thing here?

Reply:
no. did you see the changes to application.cc?

Reply:
Should we write a note that says, this function cannot be futurized.

Reply:
got it.